File: python/その他/素因数分解.py
from math import sqrt
def prime_factors(n):
    factors = []
    # 2で割り切れるだけ割り続ける
    while n % 2 == 0:
        factors.append(2)
        n = n // 2
    
    # 3, 5, 7,... と奇数で割り切れるだけ割り続ける
    for i in range(3, int(sqrt(n))+1, 2):
        while n % i == 0:
            factors.append(i)
            n = n // i
    
    # n が2より大きい場合、それ自体が素数である。
    if n > 2:
        factors.append(n)
    
    return factors

File: Nim/その他/素因数分解.nim
import math
proc prime_factors(n:int):seq[int]=
    var factors: seq[int] = @[]
    var m = n
    while m mod 2 == 0:
        factors.add(2)
        m = m div 2
    for i in countup(3,int(sqrt(float(m)))+1,2):
        while m mod i == 0:
            factors.add(i)
            m = m div i
    if m > 2:factors.add(m)
    return factors

var a:seq[seq[int]] = @[]
for i in A:a.add(prime_factors(i))